### PR TITLE
MueLu: Adding Dirichlet Boundary support to BrickAggregation

### PR DESCRIPTION
**Pull request by csiefer2**
Tests: RHEL 6

### CODE REVIEW

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 1]
```diff
@@ -77,17 +79,24 @@ namespace MueLu {
...
     SET_VALID_ENTRY("aggregation: brick y size");
     SET_VALID_ENTRY("aggregation: brick z size");
+    SET_VALID_ENTRY("aggregation: brick x Dirichlet");
+    SET_VALID_ENTRY("aggregation: brick y Dirichlet");
+    SET_VALID_ENTRY("aggregation: brick z Dirichlet");
 #undef  SET_VALID_ENTRY
 
     validParamList->set< RCP<const FactoryBase> >("A",           Teuchos::null, "Generating factory for matrix");
     validParamList->set< RCP<const FactoryBase> >("Coordinates", Teuchos::null, "Generating factory for coordinates");
+    validParamList->set< RCP<const FactoryBase> >("Graph"      , Teuchos::null, "Generating factory of the graph");
```
Comment (cgcgcg) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:89]: Are these actually used?

Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:89]: No, but they were being set anyway.

Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:89]: Fixed

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 2]
```diff
@@ -482,6 +482,33 @@
       <comment-ML>parameter not existing in ML</comment-ML>
     </parameter>
 
+   <parameter>
+      <name>aggregation: brick x Dirichlet</name>
+      <type>bool</type>
+      <default>false</default>
+      <description>Asserts that Dirichlet conditions are applied in
+        the x-direction and does not aggregate these dofs.</description>
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/masterList.xml:490]: Perhaps instead "and Dirichlet DOFs are not aggregated"

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 3]
```diff
@@ -482,6 +482,33 @@
...
+      <comment-ML>parameter not existing in ML</comment-ML>
+    </parameter>
+
+   <parameter>
+      <name>aggregation: brick y Dirichlet</name>
+      <type>bool</type>
+      <default>false</default>
+      <description>Asserts that Dirichlet conditions are applied in
+        the y-direction and does not aggregate these dofs.</description>
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/masterList.xml:499]: see above

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 4]
```diff
@@ -482,6 +482,33 @@
...
+      <comment-ML>parameter not existing in ML</comment-ML>
+    </parameter>
+
+   <parameter>
+      <name>aggregation: brick z Dirichlet</name>
+      <type>bool</type>
+      <default>false</default>
+      <description>Asserts that Dirichlet conditions are applied in
+        the z-direction and does not aggregate these dofs.</description>
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/masterList.xml:508]: see above

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp [hunk 5]
```diff
@@ -63,6 +63,7 @@
 #include "MueLu_Aggregates_fwd.hpp"
 #include "MueLu_Exceptions.hpp"
 #include "MueLu_Utilities_fwd.hpp"
+#include "MueLu_GraphBase.hpp"
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp:66]: Should the include be a forward file instead?

Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp:66]: That's not the pattern in Uncoupled, but we don't actually need it.  Fixed.

File: packages/muelu/doc/UsersGuide/masterList.xml (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:490]: got it

File: packages/muelu/doc/UsersGuide/masterList.xml (hunk 3 above)
Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:499]: see above

File: packages/muelu/doc/UsersGuide/masterList.xml (hunk 4 above)
Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:508]: see above

**Comment by csiefer2** [comment]
@trilinos/framework We're up to 7 hours on this guy.  What's going wrong?

**Comment by prwolfe** [comment]
> @trilinos/framework We're up to 7 hours on this guy. What's going wrong?

Waiting on Ride. Everything else is done.

**Pull request merged**